### PR TITLE
Fixing microcontroller.cpu on multi-core cpus and adding microcontroller.cpus

### PR DESCRIPTION
**Pull request by DavePutz**
This will allow microcontroller.cpu.temperature, etc. to work properly on an RP2040 or other multi-cpu chip. It will default to the values for cpu 0. In addition, it adds  microcontroller.cpus, indexed by cpu number, to allow access to the values for other cpus (for example, microcontroller.cpus[1].voltage)  Note that microcontroller.cpus is only added for chips where CIRCUITPY_PROCESSOR_COUNT is defined to be > 1.

### CODE REVIEW

File: shared-bindings/microcontroller/__init__.c [hunk 1]
```diff
@@ -53,7 +53,13 @@
 //| cpu: Processor
 //| """CPU information and control, such as ``cpu.temperature`` and ``cpu.frequency``
 //| (clock frequency).
-//| This object is the sole instance of `microcontroller.Processor`."""
+//| This object is an instance of `microcontroller.Processor`."""
+//|
+
+//| cpus: Processor
+//| """CPU information and control, such as ``cpus[0].temperature`` and ``cpus[1].frequency``
+//| (clock frequency) on chips with more than 1 cpu.  the index selects which cpu.
```
Comment (dhalbert) [inline shared-bindings/microcontroller/__init__.c:61]:
```suggestion
//| (clock frequency) on chips with more than 1 cpu. The index selects which cpu.
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
LGTM. Just a typo to fix.

File: shared-bindings/microcontroller/__init__.c (hunk 1 above)
Comment (DavePutz) [reply shared-bindings/microcontroller/__init__.c]: Fixed, thanks!

**Review by dhalbert — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**